<?xml version='1.0' ?>

<launch>

  <arg name="use_sim_time" default="true" description="Use the /clock topic for time to sync with simulation"/>
  <arg name="config_file" default="$(find-pkg-share roscon_demos)/config/workshop/turtlebot_config.yaml" description="The config file that provides important parameters for setting up the adapter"/>
  <arg name="nav_graph_file" default="$(find-pkg-share roscon_maps)/maps/workshop/nav_graphs/0.yaml" description="The graph that this fleet should use for navigation"/>
  <arg name="fleet_name" default="tinyRobot" description="Name of the robot fleet"/>
  <arg name="server_uri" default="" description="The URI of the api server to transmit state and task information."/>
  <arg name="output" default="screen"/>

  <!-- Fleet manager -->
  <node pkg="roscon_fleet_adapter"
        exec="fleet_manager"
        args="--config_file $(var config_file) --nav_graph $(var nav_graph_file)"
        output="both">

    <param name="use_sim_time" value="$(var use_sim_time)"/>
  </node>

  <node pkg="roscon_fleet_adapter"
        exec="fleet_adapter"
        output="both">

    <param name="use_sim_time" value="$(var use_sim_time)"/>
    <param name="nav_graph_file" value="$(var nav_graph_file)"/>
    <param name="fleet_name" value="$(var fleet_name)"/>
    <param name="server_uri" value="$(var server_uri)"/>
  </node>

</launch>
